infrastructures/kubernetes/src/main/java/org/eclipse/che/workspace/infrastructure/kubernetes/namespace/pvc/WorkspacePVCCleaner.java

```diff
@@ -56,12 +59,24 @@ public void subscribe(EventService eventService) {
           event -> {
             final String workspaceId = event.getWorkspace().getId();
             try {
-              strategy.cleanup(workspaceId);
+              if (!isEphemeral(event.getWorkspace())) {
```

Comment:
Would be better to move this logic to the strategy to avoid splitting it between components.

Reply:
yeah, good point

Reply:
@garagatyi do you think it would be  acceptable to change strategy interface e.g. `void cleanup(Workspace workspace) throws InfrastructureException;` in order to get the whole workspace config directly in strategies ?

Reply:
Yes, I do. @sleshchenko WDYT?